modules/iam/group_membership/variables.tf

```diff
@@ -0,0 +1,14 @@
+variable "name" {
```

Comment:
what is the reason we have these fields as `vars` instead of just put them in the place where we are using them? they are not secrets and they are used only once

if there is a plan to use them later in different modules, they can be put to locals of top module (bit I don't think that it is the case)

Reply:
This module was defined to reuse since we need to define the membership for 3 groups: `admins`, `developers`, and `bots`. So these variables were intended to explicitly expose the module with the needed input variables.

I think we can use it directly in the `shared/main.tf`, but IMO, it's much better and cleaner to separate it into a dedicated module.

```go
# As is 
module "iam_group_memberships" {
  source = "../modules/iam/group_memberships"

  name  = "admin-group-membership"
  group = module.iam_groups.admin_group
  users = local.iam_admin_emails
}

# To be
resource "aws_iam_group_membership" "admin_group_membership" {
  name  = "admin-group-membership"
  group = module.iam_groups.admin_group
  users = local.iam_admin_emails
}
```

Reply:
It sounds reasonable to leave it as is